plugins/pulp_rpm/plugins/importers/yum/existing.py

```diff
@@ -135,14 +135,15 @@ def check_all_and_associate(wanted, conduit, config, download_deferred, catalog)
                 # package file does not exist and downloading is not deferred.
                 if not download_deferred and not file_exists:
                     continue
-            catalog.add(unit, wanted[unit.unit_key_as_named_tuple].download_path)
-            if rpm_parse.signature_enabled(config):
-                try:
-                    rpm_parse.filter_signature(unit, config)
-                except PulpCodedException as e:
-                    _LOGGER.debug(e)
-                    continue
-            repo_controller.associate_single_unit(conduit.repo, unit)
+            if not unit.is_associated(conduit.repo.repo_id):
```

Comment:
For the same 14K rpms repo this improvement give us minus 30 sec or so , which is 2-3% of the sync task. I pushed the change, please re-review @mhrivnak